salt/metalk8s/orchestrate/deploy_node.sls

```diff
@@ -113,6 +113,35 @@ Wait minion available:
 
 {%- endif %}
 
+{%- if 'etcd' in pillar.get('metalk8s', {}).get('nodes', {}).get(node_name, {}).get('roles', []) %}
+
+# Prepare etcd and dependencies
+Prepare etcd dependencies:
+  salt.state:
+    - tgt: {{ node_name }}
+    - saltenv: metalk8s-{{ version }}
+    - sls: metalk8s.roles.etcd.prepared
+    - require:
+      - salt: Set grains
+      - salt: Refresh the mine
+      - salt: Refresh and check pillar before highstate
+
+Register the node into etcd cluster:
+  salt.runner:
+    - name: state.orchestrate
+    - pillar: {{ pillar | json  }}
+    - mods:
+      - metalk8s.orchestrate.register_etcd
+    - require:
+      - salt: Set grains
+      - salt: Refresh the mine
+      - salt: Refresh and check pillar before highstate
+      - salt: Prepare etcd dependencies
+    - require_in:
+      - salt: Run the highstate
```

Comment:
Maybe add `metalk8s.kubernetes.etcd` between registering and highstate, to make sure `etcd` is unavailable the least amount of time possible. WDYT @NicolasT @TeddyAndrieux ?

Reply:
I think/hope that etcd cluster work when a node get registered but not joined. So IMO no.

Reply:
> etcd cluster work when a node get registered but not joined

No it does not

Reply:
When going from 1 to 2 nodes indeed it does not, so we want to make the "time" during which 2 nodes are registered but only 0 or 1 of them are running as short as possible, i.e., apply an as-specific-state as possible.

Reply:
Then I suggest we use the `skip_roles` so, we first run highstate without `etcd` and then we do etcd steps in the right order